### PR TITLE
Add typescript check to Makefile 

### PR DESCRIPTION
**Pull request by brainopia**
Previous discussion in #453 
If there is anything you'd like me to change, just say the word :)

---
This change is [Reviewable](https://reviewable.io/reviews/witheve/eve/454)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,20 +14,9 @@ Eve is a set of tools to help us think. Currently, these tools include: a tempor
 
 ### From Source
 
-Before you can install Eve you'll need need LuaJIT, gcc, make, and python. Currently, building from source is only supported on Linux and OSX. On Windows, we've managed to compile Eve in cygwin, but have not gotten it to run yet. Contributions are welcome on this front.
```
Comment (shamrin) [inline README.md]: Why stop mentioning LuaJIT?

Comment (brainopia) [inline README.md]: Users don't need to install LuaJIT manually anymore. It's built by default with Eve.

File: README.md [hunk 2]
```diff
@@ -14,20 +14,9 @@ Eve is a set of tools to help us think. Currently, these tools include: a tempor
...
-make
-make install
-```
-
-By default, LuaJIT is not added to your path, so you'll need to do that as well:
-
-```
-ln -sf luajit-2.1.0-beta2 /usr/local/bin/luajit
-```
+First install gcc, make, and npm using your operating system's standard channels. Chances are you already have these.
```
Comment (tbodt) [inline README.md]: NPM currently isn't preinstalled on very many operating systems...

File: build/Makefile [hunk 3]
```diff
@@ -70,17 +67,20 @@ contgen: contgen.c
...
 	cc -std=c99 -c -g -I$(LUAJIT_INCLUDES) $^
 
-
-%.js: $(BUILD)/../jssrc/%.ts
-	cd .. && tsc --pretty
+%.js: ../jssrc/%.ts
+	@# check if there is an appropriate typescript installation
+	@# otherwise install it locally with npm
+	@$(WITH_NPM_PATH) tsc -v | grep -q 'Version 2' || \
+		npm install typescript || (echo 'Please install npm' && exit 1)
```
Comment (joshuafcole) [inline build/Makefile]: I think if we're going to go the local tsc route installed via npm we ought to use use a regular package.json and automatically invoke `npm i` at the beginning of the makefile. That way we get semver and other tooling benefits for free.

Comment (joshuafcole) [inline build/Makefile]: If you want to update the PR to do that I can throw together a package.json, or you're welcome to generate a skeletal one with typescript@2.0

Comment (brainopia) [inline build/Makefile]: Sure, go ahead, thanks :+1:

**Pull request closed without merging**